houndigrade/test_cli.py

```diff
@@ -0,0 +1,251 @@
+
+    @patch('cli.Connection')
+    @patch('cli.glob.glob')
+    @patch('cli.subprocess.run')
+    def test_cli_disappearing_files(self, mock_subprocess_run, mock_glob_glob,
+                                    mock_connection):
+        cloud = 'aws'
+        image_id = 'ami-123456789'
+        drive_path = './dev/xvdf'
+
+        def mock_glob_side_effect(pattern):
+            if 'etc/*-release' in pattern:
+                return ['./dev/xvdf/xvdf1/etc/redhat-release',
+                        './dev/xvdf/xvdf1/etc/os-release',
+                        './dev/xvdf/xvdf2/etc/centos-release',
+                        './dev/xvdf/xvdf2/etc/os-release', ]
+            else:
+                return ['./dev/xvdf1', './dev/xvdf2', ]
+
+        mock_glob_glob.side_effect = mock_glob_side_effect
+
+        mock_with_conn = mock_connection.__enter__.return_value
+        mock_producer = mock_with_conn.Producer.return_value
+        mock_pub = mock_producer.publish
+        mock_pub.return_value = True
+
+        runner = CliRunner()
+
+        with runner.isolated_filesystem():
+            pathlib.Path('{}/xvdf1'.format(drive_path)).mkdir(parents=True,
+                                                              exist_ok=True)
+            result = runner.invoke(main,
+                                   ['-c', cloud, '--debug', '-t', image_id,
+                                    drive_path])
+
+        self.assertTrue(mock_subprocess_run.called)
+        self.assertEqual(result.exit_code, 0)
+        self.assertIn(' No such file or directory', result.output)
+
+    @patch('cli.Connection')
+    @patch('cli.glob.glob')
+    @patch('cli.subprocess.run')
+    def test_cli_no_version_files(self, mock_subprocess_run, mock_glob_glob,
+                                  mock_connection):
+        cloud = 'aws'
+        image_id = 'ami-123456789'
+        drive_path = './dev/xvdf'
+
+        def mock_glob_side_effect(pattern):
+            if 'etc/*-release' in pattern:
+                return []
+            else:
+                return ['./dev/xvdf1', './dev/xvdf2', ]
+
+        mock_glob_glob.side_effect = mock_glob_side_effect
+
+        mock_with_conn = mock_connection.__enter__.return_value
+        mock_producer = mock_with_conn.Producer.return_value
+        mock_pub = mock_producer.publish
+        mock_pub.return_value = True
+
+        runner = CliRunner()
+
+        with runner.isolated_filesystem():
+            pathlib.Path('{}/xvdf1'.format(drive_path)).mkdir(parents=True,
+                                                              exist_ok=True)
+            result = runner.invoke(main,
+                                   ['-c', cloud, '--debug', '-t', image_id,
+                                    drive_path])
+
+        self.assertTrue(mock_subprocess_run.called)
+        self.assertEqual(result.exit_code, 0)
+        self.assertIn(
+            '"status": "No release files found on ./dev/xvdf1"', result.output)
```

Comment:
You should construct the "No release files found on" string using `_` like it is in the main code because if someone runs the tests on a non-English-localized system, that could cause this test to fail.

Reply:
Or we force the English localization on our tests! I'm okay with either solution, although I don't know off-hand how to conveniently force English in the scope of a test.